File: WinForms/DevExpress.OutlookInspiredApp.Win/VB/DevExpress.OutlookInspiredApp.Win/Modules/Opportunities/QuotesFilterPaneCollapsed.vb
Imports DevExpress.DevAV.Common.Utils
Imports DevExpress.DevAV.Presenters
Imports DevExpress.DevAV.ViewModels
Imports DevExpress.XtraLayout.Utils

Namespace DevExpress.DevAV.Modules

    Public Partial Class QuotesFilterPaneCollapsed
        Inherits BaseModuleControl
        Implements ISupportCompactLayout

        Private presenterCore As QuoteFilterPanePresenter

        Public Sub New(ByVal collectionViewModel As QuoteCollectionViewModel)
            MyBase.New(GetType(QuotesFilterTreeViewModel), New Object() {collectionViewModel})
            InitializeComponent()
            presenterCore = CreatePresenter()
            BindCommands()
            SwapFluentIcons(Me)
        End Sub

        Protected Overrides Sub OnDisposing()
            Presenter.Dispose()
            MyBase.OnDisposing()
        End Sub

        Public ReadOnly Property ViewModel As QuotesFilterTreeViewModel
            Get
                Return GetViewModel(Of QuotesFilterTreeViewModel)()
            End Get
        End Property

        Public ReadOnly Property Presenter As QuoteFilterPanePresenter
            Get
                Return presenterCore
            End Get
        End Property

        Protected Overridable Function CreatePresenter() As QuoteFilterPanePresenter
            Return New QuoteFilterPanePresenter(navigationBar, ViewModel)
        End Function

        Protected Overridable Sub BindCommands()
            btnNew.BindCommand(Sub() Presenter.CollectionViewModel.[New](), Presenter.CollectionViewModel)
        End Sub

#Region "ISupportCompactLayout Members"
        Private compactLayout As Boolean = True

        Private Property Compact As Boolean Implements ISupportCompactLayout.Compact
            Get
                Return compactLayout
            End Get

            Set(ByVal value As Boolean)
                If compactLayout = value Then Return
                compactLayout = value
                UpdateCompactLayout()
            End Set
        End Property

        Private Sub UpdateCompactLayout()
            btnNew.Visible = Not compactLayout
        End Sub
#End Region
    End Class
End Namespace

File: WinForms/DevExpress.OutlookInspiredApp.Win/CS/DevExpress.OutlookInspiredApp.Win/Modules/Opportunities/QuotesFilterPaneCollapsed.cs
namespace DevExpress.DevAV.Modules {
    using DevExpress.DevAV.Common.Utils;
    using DevExpress.DevAV.Presenters;
    using DevExpress.DevAV.ViewModels;
    using DevExpress.XtraLayout.Utils;

    public partial class QuotesFilterPaneCollapsed : BaseModuleControl, ISupportCompactLayout {
        QuoteFilterPanePresenter presenterCore;
        public QuotesFilterPaneCollapsed(QuoteCollectionViewModel collectionViewModel)
            : base(typeof(QuotesFilterTreeViewModel), new object[] { collectionViewModel }) {
            InitializeComponent();
            presenterCore = CreatePresenter();
            BindCommands();
            FluentIconHelper.SwapFluentIcons(this);
        }
        protected override void OnDisposing() {
            Presenter.Dispose();
            base.OnDisposing();
        }
        public QuotesFilterTreeViewModel ViewModel {
            get { return GetViewModel<QuotesFilterTreeViewModel>(); }
        }
        public QuoteFilterPanePresenter Presenter {
            get { return presenterCore; }
        }
        protected virtual QuoteFilterPanePresenter CreatePresenter() {
            return new QuoteFilterPanePresenter(navigationBar, ViewModel);
        }
        protected virtual void BindCommands() {
            btnNew.BindCommand(() => Presenter.CollectionViewModel.New(), Presenter.CollectionViewModel);
        }
        #region ISupportCompactLayout Members
        bool compactLayout = true;
        bool ISupportCompactLayout.Compact {
            get { return compactLayout; }
            set {
                if(compactLayout == value) return;
                compactLayout = value;
                UpdateCompactLayout();
            }
        }
        void UpdateCompactLayout() {
            btnNew.Visible = !compactLayout;
        }
        #endregion
    }
}
